packages/block-library/src/template-part/edit/index.js

```diff
@@ -23,7 +23,11 @@ export default function TemplatePartEdit( {
 	const initialTheme = useRef( theme );
 
 	// Resolve the post ID if not set, and load its post.
-	const postId = useTemplatePartPost( _postId, slug, theme );
+	const postId = useTemplatePartPost(
+		_postId,
+		initialSlug.current,
+		initialTheme.current
+	);
```

Comment:
In `useTemplatePartPost` the slug and theme are only used where no ID is available.  It only makes sense to use the initialSlug and initialTheme values for this hook, as using the current `slug` attribute was causing the template part to disappear when renamed before saving and having an Id.

Reply:
Reverted this as part of the new changes.